src/main/java/com/training/web/pages/flipkart/FlipkartHomePage.java

```diff
@@ -25,4 +32,25 @@ public ResultPage clickSearch() {
         $x(submit).shouldBe(visible).click();
         return new ResultPage();
     }
+
+
+    public String clickSocialMediaPage(String media) {
+        $x("//a[text()='"+media+"']").shouldBe(visible).scrollIntoView(true).click();
+
+        switch (media) {
+            case "Facebook":
+                $x(facebookIcon).shouldBe(enabled);
+                break;
+
+            case "Twitter":
+                $x(twitterIcon).shouldBe(enabled);
+                break;
+
+            case "YouTube":
+                $x(youtubeIcon).shouldBe(enabled);
+                break;
+
+        }
+        return driver.getCurrentUrl();
```

Comment:
As per the function name, it should only click on the social media url.
Getting the current url -> do it directly on the test class with the assertions.